pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -0,0 +1,343 @@
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Deployment",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      GetSplunkMonitoringConsoleDeploymentName(instanceType, cr.GetNamespace()),
+			Namespace: cr.GetNamespace(),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: selectLabels,
+			},
+			Replicas: &replicas,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:      labels,
+					Annotations: annotations,
+				},
+				Spec: corev1.PodSpec{
+					Affinity:      affinity,
+					Tolerations:   spec.Tolerations,
+					SchedulerName: spec.SchedulerName,
+					Containers: []corev1.Container{
+						{
+							Image:           spec.Image,
+							ImagePullPolicy: corev1.PullPolicy(spec.ImagePullPolicy),
+							Name:            "splunk",
+							Ports:           ports,
+							EnvFrom: []corev1.EnvFromSource{
+								{
+									ConfigMapRef: &corev1.ConfigMapEnvSource{
+										LocalObjectReference: corev1.LocalObjectReference{
+											Name: configMap, //monitoring console env variables configMap
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if configMapHash == "" {
+		return nil, nil
+	}
+
+	env := []corev1.EnvVar{
+		{Name: "SPLUNK_CONFIGMAP_HASH",
+			Value: configMapHash,
+		},
+	}
+	// append labels and annotations from parent
+	splcommon.AppendParentMeta(deploymentMC.Spec.Template.GetObjectMeta(), cr.GetObjectMeta())
+
+	// update statefulset's pod template with common splunk pod config
+	updateSplunkPodTemplateWithConfig(&deploymentMC.Spec.Template, cr, spec, instanceType, env, secretName)
+
+	return deploymentMC, nil
+}
+
+//ApplyMonitoringConsoleEnvConfigMap creates or updates a Kubernetes ConfigMap for extra env for monitoring console pod
+func ApplyMonitoringConsoleEnvConfigMap(client splcommon.ControllerClient, namespace string, crName string, newURLs []corev1.EnvVar, addNewURLs bool) (*corev1.ConfigMap, error) {
+
+	var current corev1.ConfigMap
+	current.Data = make(map[string]string)
+
+	//1. Retrieve the existing McExtraEnvMap contents
+	namespacedName := types.NamespacedName{Namespace: namespace, Name: GetSplunkMonitoringconsoleConfigMapName(namespace, SplunkMonitoringConsole)}
+	err := client.Get(context.TODO(), namespacedName, &current) //save existing in current
+
+	//If no configMap and deletion of CR is requested then retrun nil
+	if err != nil && addNewURLs == false {
+		//create a configMap with no values
+		current = corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      GetSplunkMonitoringconsoleConfigMapName(namespace, SplunkMonitoringConsole),
+				Namespace: namespace,
+			},
+			Data: make(map[string]string),
+		}
+
+		err = splctrl.CreateResource(client, &current)
+		if err != nil {
+			return nil, err
+		}
+		return &current, nil
+	}
+
+	if err == nil {
+		revised := current.DeepCopy()
+		for _, url := range newURLs {
```

Comment:
Can this for loop and the string operations inside be divided into functions? To make it clearer?

Reply:
That's TODO there, will have a short story to improvise this logic